scanner/src/main/kotlin/scanners/fossid/FossId.kt

```diff
@@ -447,11 +460,28 @@ class FossId internal constructor(
             }
 
             checkScan(scanCode, *deltaScanRunParameters(existingScancode))
+
+            enforceDeltaScanLimit(recentScans)
         }
 
         return scanCode
     }
 
+    /**
+     * Make sure that only the configured number of delta scans exists for the current package. Based on the list of
+     * [existingScans], delete older scans until the maximum number of delta scans is reached.
+     */
+    private suspend fun enforceDeltaScanLimit(existingScans: List<Scan>) {
+        log.info { "Number of delta scans to keep: ${config.deltaScanLimit}." }
+        // The current scan needs to be counted as well, in addition to the already existing scans.
+        if (existingScans.size + 1 > config.deltaScanLimit) {
+            log.info { "Deleting ${existingScans.size - config.deltaScanLimit + 1} older scans." }
+        }
+
+        existingScans.drop(config.deltaScanLimit - 1)
```

Comment:
`drop()` "Returns a list containing all elements except first *n* elements", so it drops the *n* most *recent* scans. But don't we want to drop / delete the *oldest* scans that exceed the limit?

Reply:
This is a bit tricky, maybe a comment is needed here as well. From the list the _n_ most recent scans are removed, but then the *remaining* ones are deleted. This basically means that the most recent scans are kept.

Reply:
Ah, sorry, mindfart on my side. So yes, maybe saying some like `// Drop the scans to keep in order to iterate over the remaining ones to delete them.` would help.

Reply:
Added a comment.